sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/AvailabilityStrategy.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.util.List;
+
+/**
+ * The type Availability strategy.
+ */
+public abstract class AvailabilityStrategy {
+    /**
+     * The Effective retry regions.
+     */
+    /**
+     * The Number of regions to try.
+     */
+    int numberOfRegionsToTry;
+
+    /**
+     * Gets effective retry regions. This API also gives the preferred regions configured and the regions to exclude
+     * from the preferred regions. Use this to specify the regions to retry on.
+     *
+     * @param preferredRegions the preferred regions
+     * @param excludeRegions   the regions to exclude from the preferred regions
+     * @return the effective retry regions
+     */
+    public abstract List<String> getEffectiveRetryRegions(List<String> preferredRegions, List<String> excludeRegions);
```

Comment:
how this method will be used by customer? excludeRegions can be defined at a different time than when the strategy being defined. I think this belong to internal/implementation

Reply:
+1

Reply:
Removed this after refactoring that logic